### PR TITLE
creating new release for 2.4.0

### PR DESCRIPTION
**Pull request by erezz-mov-ai**
creating new release for 2.4.0

### CODE REVIEW

**Review by engdorm — CHANGES_REQUESTED** [review]
I'm not sure bout the deletion requirements.txt

File: flow_initiator/spawner/flowmonitor.py [hunk 1]
```diff
@@ -465,7 +465,7 @@ def get_node_type_cmd(self, node_name: str, flow: Flow) -> list:
         elif node_type == ROS1_PLUGIN:
             output = []
         else:
-            output = ["/usr/bin/python3", f"{APP_PATH}/GD_Node.py"]
+            output = ["gd_node"]
```
Comment (AlexFernandes-MOVAI) [inline flow_initiator/spawner/flowmonitor.py:468]: so, this `gd_node` binary needs to be in the path  ?

**Review by AlexFernandes-MOVAI — COMMENTED** [review]
OK, so here you create a branch for the future 2.4.0 release :
- main/dev : hold the 2.3.x release line
- develop/2.4.0 and releases/2.4.0 : hold the 2.4.x release line
This is valid, just make sure to:
- communicate about this
- merge back develop/2.4.0 and releases/2.4.0 as soon as 2.4.0 becomes the official latest version

**Review by AlexFernandes-MOVAI — CHANGES_REQUESTED** [review]
please align the version number of branches and modules:
- if branches are called something/2.4.0, the modules should be versioned 2.4.x

File: flow_initiator/spawner/flowmonitor.py (hunk 1 above)
Comment (engdorm) [reply flow_initiator/spawner/flowmonitor.py:468]: the gd_node will be an endpoint inside the gd-node python package

Comment (erezz-mov-ai) [reply flow_initiator/spawner/flowmonitor.py:468]: yes it was added to the setup.py as an entry point

**Comment by erezz-mov-ai** [comment]
> I'm not sure bout the deletion requirements.txt



>

**Comment by erezz-mov-ai** [comment]
> I'm not sure bout the deletion requirements.txt

the requirements are mentioned inside the setup.py, so requirements.txt is not longer needed.
also the pipline don't know how to handle it.

File: flow_initiator/spawner/flowmonitor.py (hunk 1 above)
Comment (erezz-mov-ai) [reply flow_initiator/spawner/flowmonitor.py:468]: yes we must extend update PATH env to include:
/opt/mov.ai/.local/bin

**Review by engdorm — APPROVED** [review]

**Pull request merged**